server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/DatafileManager.java

```diff
@@ -69,13 +70,16 @@
   // ensure we only have one reader/writer of our bulk file notes at at time
   private final Object bulkFileImportLock = new Object();
 
-  // This must be incremented whenever datafileSizes is mutated
-  private long updateCount;
+  // This must be incremented before and after datafileSizes and metadata table updates. These
+  // counts allow detection of overlapping operation w/o placing a lock around metadata table
```

Comment:
I _think_ I know what you are trying to say with "overlapping operation" but maybe you could expand on it with an additional sentence? Or maybe reword "overlapping operation" to something else?

Reply:
I made multiple comment updates in 7d6b61e

Reply:
Thanks, those are a BIG help!